### ISSUE TITLE
[Support]: Car/person detection during the day and only people at night?

### ISSUE THREAD

**Issue by junior466** [blakeblackshear/frigate#2876]
### Describe the problem you are having

I am only opening this as an issue because with the upcoming launch of Frigate+, I suspect this to be something people may request or want to achieve. 

I noticed that the way my camera is pointed is my driveway always gives me false alerts with parked cars so along with the power of Home Assistant, I configured my detectors to only be on when we are away and know that the cars are not there. I achieved that using location and a simple automation. The issue now is at night, I would like to turn on the detector but only allow the detector to pick up people, no cars as I know the cars will be there in the zone causing false alerts. 

Is this possible? For some reason, I can't think it through. 

### Version

0.10.0

### Frigate config file

```yaml
cameras:
  driveway_cam:
    rtmp:
      enabled: False

    ffmpeg:
      inputs:
        - path: rtsp://ha:REDACTED$@192.168.141.14:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - detect
            - record

    detect:
      width: 3840
      height: 2160
      fps: 5
      enabled: True

    zones:
      driveway:
        coordinates: 2876,2160,2083,1823,1928,1413,2175,1308,3840,1746,3840,2160
        objects:
          - person
          - dog
          - car

    objects:
      track:
        - person
        - dog
        - car
      filters:
        person:
          # Optional: minimum width*height of the bounding box for the detected object (default: 0)
          min_area: 5000
          # Optional: maximum width*height of the bounding box for the detected object (default: 24000000)
          max_area: 100000
          mask:
            - 0,0,0,2160,2098,2160,1928,1374,2690,948,3840,1130,3840,0
        car:
          # Optional: minimum width*height of the bounding box for the detected object (default: 0)
          min_area: 5000
          # Optional: maximum width*height of the bounding box for the detected object (default: 24000000)
          max_area: 200000
          mask:
            - 0,0,0,2160,2098,2160,1928,1374,2690,948,3840,1130,3840,0
        dog:
          mask:
            - 0,0,0,2160,2098,2160,1928,1374,2690,948,3840,1130,3840,0

    motion:
      # Increasing this value will make motion detection less sensitive
      # threshold: 30
      # default 25

      # Minimum size in pixels in the resized motion image that counts as motion.
      # Increasing this value will prevent smaller areas of motion from being detected. Decreasing will
      # make motion detection more sensitive to smaller moving objects.
      # As a rule of thumb:
      #  - 15 - high sensitivity
      #  - 30 - medium sensitivity
      #  - 50 - low sensitivity
      contour_area: 35
      # default: 30

      mask:
       - 0,0,0,2160,2098,2160,1928,1374,2690,948,3840,1130,3840,0

    record:
      enabled: True
      retain:
        days: 7
        mode: motion
      events:
        max_seconds: 240
        pre_capture: 5
        post_capture: 5
        retain:
          default: 7
          mode: active_objects
        objects:
          - person
          - car
          - dog
        required_zones:
          - driveway

    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      crop: False
      required_zones:
        - driveway
      retain:
        default: 7
#####

mqtt:
  host: 192.168.141.21
  user: REDACTED
  password: REDACTED

detectors:
  coral:
    type: edgetpu
    device: usb

database:
  path: /db/frigate.db

snapshots:
  enabled: True
  timestamp: True
  bounding_box: True
  crop: False
  required_zones: []
  retain:
    default: 7

model:
  labelmap:
    15: animal
    16: animal
    17: animal
    18: animal
    19: animal
    20: animal
    21: animal
    22: animal
    23: animal
    24: animal

birdseye:
  enabled: False
  # width: 1920
  # height: 1080
  # quality: 8
  # mode: motion
```

### Relevant log output

```shell
NA
```

### FFprobe output from your camera

```shell
NA
```

### Operating system

Debian

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

Amcrest 4K

**Comment by NickM-27** [comment]
It is possible to ignore it in home assistant but events will still be created. This has been requested previously and I believe there is a similar pinned issue.

**Comment by junior466** [comment]
Actually, I think there's no need for that. Simply creating another automation that only notifies when a person is detected and enabling it at night may be the way to go.

**Comment by junior466** [comment]
@NickM-27 and @blakeblackshear  Exactly that!

**Comment by NickM-27** [comment]
To be clear, not complaining as I know night is particularly challenging not to mention I have a feeling the COCO dataset doesn't have much data with night vision shots.

I personally just have the automation filter cars after a certain time. As that seems much more prone to night time false positives than person

**Comment by junior466** [comment]
> To be clear, not complaining as I know night is particularly challenging not to mention I have a feeling the COCO dataset doesn't have much data with night vision shots.
> 
> I personally just have the automation filter cars after a certain time. As that seems much more prone to night time false positives than person

So you have an automation that at certain times, disables notifications? Just fishing for ideas ;-)

**Comment by NickM-27** [comment]
I have a single automation. Says if object label is car then it must be before sunset to send notifications, other labels always send notifications

**Comment by junior466** [comment]
And with the above, you will see many events of your parked car I assume?

**Comment by blakeblackshear** [comment]
Does your parked car score too low to be tracked as a stationary object at night?

**Comment by NickM-27** [comment]
@blakeblackshear In my case it is possible that some camera settings might help, I haven't tried adjusting that, but the car is barely visible to the camera until the light from another car lights it up. 

![Screen Shot 2022-02-26 at 7 51 44 PM](https://user-images.githubusercontent.com/14866235/155866212-6949c836-fb00-47f7-844a-f9327167664d.png)

**Comment by blakeblackshear** [comment]
It sounds like both of you have zones in the driveway with required zones set on events/snapshots/notifications. It may just be that you are asking to be notified when cars are in your driveway. Have you tried defining a zone for the street and requiring that cars have entered more than 1 zone for notifications? That should only notify you for cars that enter the driveway after having been on the street.

**Comment by NickM-27** [comment]
> It sounds like both of you have zones in the driveway with required zones set on events/snapshots/notifications. It may just be that you are asking to be notified when cars are in your driveway. Have you tried defining a zone for the street and requiring that cars have entered more than 1 zone for notifications? That should only notify you for cars that enter the driveway after having been on the street. 

Hmm, that's a smart idea, that way I could keep the stationary tracking during the day and also keep the notifications for when my fiancé gets home after dark without worrying about the notifications from what we're talking about here. Will have to try that, and maybe tweaking the camera settings 👌

**Comment by junior466** [comment]
> Have you tried defining a zone for the street and requiring that cars have entered more than 1 zone for notifications? That should only notify you for cars that enter the driveway after having been on the street.

That sound like a great idea but I am not sure how to implement it. My street is visible from the camera and is quite active. Do it create a zone for the actual street and one for my driveway?

**Comment by blakeblackshear** [comment]
Yes. The street would be a separate zone.

**Comment by junior466** [comment]
I see. I think it makes sense. A zone for the ‘street’ and one for ‘driveway’. Under recording and snapshot, only have ‘driveway’ for required otherwise my busy street would be recorded/snapshotted all the time. Under iOS notifications, the required zone would be both?

**Comment by junior466** [comment]
> Does your parked car score too low to be tracked as a stationary object at night?

And to answer this earlier question, yes it does.

**Comment by junior466** [comment]
Well, I created the two zones (street and driveway) and made sure to create a notification requiring both zones but I am always alerted when a car enters the `street` zone. Based on your comment `Have you tried defining a zone for the street and requiring that cars have entered more than 1 zone for notifications? `, I was under the impression that both zones would need to be triggered for a notification?

Am I missing something obvious? And thank you for your patience and support.

**Comment by junior466** [comment]
Sorry, forgot to attach a screenshot:

![Screen Shot 2022-02-28 at 8 46 44 AM](https://user-images.githubusercontent.com/22748735/155994863-104d0e0a-44e5-4d0e-8fc2-f2d5a19280d7.png)

**Comment by NickM-27** [comment]
@junior466 How did you setup your notification, are you using the notification blueprint? I believe the notification blueprint ORs the zones. May need to set something up yourself for that.

**Comment by junior466** [comment]
Yes, I am using the Blueprint. That makes sense. I need to do some research on creating my own notification. By any chance, are you using something similar that you can share? If not, no big deal.

**Comment by NickM-27** [comment]
@junior466 Keep in mind this is decently customized to me, and I have my own script which sends notifications to my devices, but this should be a good starting point.

```yaml
alias: Outside - Driveway & Front Yard - Frigate Event - Cat / Dog / Person Seen
description: ''
trigger:
  - platform: mqtt
    topic: frigate/events
    id: frigate-event
    payload: front_doorbell_cam
    value_template: '{{ value_json["after"]["camera"] }}'
condition:
  - condition: or
    conditions:
      - condition: template
        value_template: '{{ trigger.payload_json[''type''] == ''new'' }}'
      - condition: template
        value_template: '{{ trigger.payload_json["before"]["entered_zones"]|length == 0 }}'
  - condition: template
    value_template: '{{ trigger.payload_json["after"]["entered_zones"]|length > 0 }}'
  - condition: template
    value_template: >-
      {{ ["driveway", "front_yard"]|select("in",
      trigger.payload_json["after"]["entered_zones"])|list|length > 0 }}
  - condition: template
    value_template: '{{ trigger.payload_json["after"]["label"] != "car" }}'
action:
  - choose:
      - conditions:
          - condition: trigger
            id: frigate-event
        sequence:
          - service: script.send_dynamic_notification_to_all_devices
            data:
              id: '{{ trigger.payload_json["after"]["id"] }}'
              title: >-
                {{ trigger.payload_json["after"]["label"] | title }} was
                detected on {{ trigger.payload_json["after"]["camera"] |
                replace("_", " ") | title }}
              message: >-
                {{ trigger.payload_json["after"]["label"] | title }} detected in
                the {{ trigger.payload_json["after"]["entered_zones"][0] }}
              notifTag: '{{ trigger.payload_json["after"]["id"] }}'
              notifIcon: mdi:doorbell-video
              group: >-
                frigate-notification-{{ trigger.payload_json["after"]["camera"]
                }}
              importance: max
              image: >-
                /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
              clickAction: ''
          - repeat:
              until:
                - condition: template
                  value_template: '{{ wait.trigger.payload_json["type"] == "end" }}'
              sequence:
                - wait_for_trigger:
                    - platform: mqtt
                      topic: frigate/events
                      payload: '{{ trigger.payload_json["after"]["id"] }}'
                      value_template: '{{ value_json["after"]["id"] }}'
                  continue_on_timeout: false
                  timeout: '00:02:00'
                - condition: template
                  value_template: '{{ wait.trigger.payload_json[''type''] == ''end'' }}'
                - service: script.send_dynamic_notification_to_all_devices
                  data:
                    id: '{{ trigger.payload_json["after"]["id"] }}'
                    title: >-
                      {% if (state_attr("sensor.double_take_front_doorbell_cam",
                      "id") == trigger.payload_json["after"]["id"]) and
                      (state_attr("sensor.double_take_front_doorbell_cam",
                      "matches") | length > 0) -%}
                      {{(state_attr("sensor.double_take_front_doorbell_cam",
                      "matches") | first)["name"] | title}} {%- else -%}
                      {{trigger.payload_json["after"]["label"] | title}} {%-
                      endif %} was detected on {{
                      trigger.payload_json["after"]["camera"] | replace("_", "
                      ") | title }}
                    message: >-
                      {% if (state_attr("sensor.double_take_front_doorbell_cam",
                      "id") == trigger.payload_json["after"]["id"]) and
                      (state_attr("sensor.double_take_front_doorbell_cam",
                      "matches") | length > 0) -%}
                      {{(state_attr("sensor.double_take_front_doorbell_cam",
                      "matches") | first)["name"] | title}} {%- else -%}
                      {{trigger.payload_json["after"]["label"] | title}} {%-
                      endif %} detected in the {{
                      trigger.payload_json["after"]["entered_zones"][0] }}
                    notifTag: '{{ trigger.payload_json["after"]["id"] }}'
                    notifIcon: mdi:doorbell-video
                    group: >-
                      frigate-notification-{{
                      trigger.payload_json["after"]["camera"] }}
                    importance: default
                    image: >-
                      /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                    clickAction: >-
                      https://storagewhacker.homeassistantlawn.xyz/api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
    default: []
mode: parallel
max: 10

```

**Comment by NickM-27** [comment]
You'll want to adjust the 

```yaml
- condition: template
   value_template: '{{ trigger.payload_json["before"]["entered_zones"]|length == 0 }}'
```

To be something that checks for driveway in `current_zones` and street in `entered_zones`

**Comment by junior466** [comment]
I see, thank you very much. I will give this a try.

**Comment by junior466** [comment]
So as I never been good with template, do you mind a little extra assistance? 😬 I am not too sure on what values to edit. As suggested above, I am simply trying to configure a notification for when a car enters the ‘street’ and ‘driveway’ zone I receive a notification on my iOS device.

**Comment by NickM-27** [comment]
@junior466 Keep in mind this is decently customized to me, and I have my own script which sends notifications to my devices, but this should be a good starting point.

```yaml
alias: Outside - Driveway & Front Yard - Frigate Event - Cat / Dog / Person Seen
description: ''
trigger:
  - platform: mqtt
    topic: frigate/events
    id: frigate-event
    payload: NEED TO UPDATE NAME
    value_template: '{{ value_json["after"]["camera"] }}'
condition:
  - condition: or
    conditions:
      - condition: template
        value_template: '{{ trigger.payload_json[''type''] == ''new'' }}'
      - condition: template
        value_template: '{{ trigger.payload_json["before"]["entered_zones"]|length == 0 }}'
  - condition: template
    value_template: '{{ ["street"]|select("in", trigger.payload_json["after"]["entered_zones"])|list|length > 0 and ["driveway"]|select("in", trigger.payload_json["after"]["current_zones"])|list|length > 0 }}'
  - condition: template
    value_template: >-
      {{ ["driveway", "front_yard"]|select("in",
      trigger.payload_json["after"]["entered_zones"])|list|length > 0 }}
  - condition: template
    value_template: '{{ trigger.payload_json["after"]["label"] != "car" }}'
action:
  - choose:
      - conditions:
          - condition: trigger
            id: frigate-event
        sequence:
          - service: script.send_dynamic_notification_to_all_devices
            data:
              id: '{{ trigger.payload_json["after"]["id"] }}'
              title: >-
                {{ trigger.payload_json["after"]["label"] | title }} was
                detected on {{ trigger.payload_json["after"]["camera"] |
                replace("_", " ") | title }}
              message: >-
                {{ trigger.payload_json["after"]["label"] | title }} detected in
                the {{ trigger.payload_json["after"]["entered_zones"][0] }}
              notifTag: '{{ trigger.payload_json["after"]["id"] }}'
              notifIcon: mdi:doorbell-video
              group: >-
                frigate-notification-{{ trigger.payload_json["after"]["camera"]
                }}
              importance: max
              image: >-
                /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
              clickAction: ''
          - repeat:
              until:
                - condition: template
                  value_template: '{{ wait.trigger.payload_json["type"] == "end" }}'
              sequence:
                - wait_for_trigger:
                    - platform: mqtt
                      topic: frigate/events
                      payload: '{{ trigger.payload_json["after"]["id"] }}'
                      value_template: '{{ value_json["after"]["id"] }}'
                  continue_on_timeout: false
                  timeout: '00:02:00'
                - condition: template
                  value_template: '{{ wait.trigger.payload_json[''type''] == ''end'' }}'
                - service: script.send_dynamic_notification_to_all_devices
                  data:
                    id: '{{ trigger.payload_json["after"]["id"] }}'
                    title: >-
                      {{trigger.payload_json["after"]["label"] | title}} was detected on {{
                      trigger.payload_json["after"]["camera"] | replace("_", "
                      ") | title }}
                    message: >-
                      {{trigger.payload_json["after"]["label"] | title}} detected in the {{
                      trigger.payload_json["after"]["entered_zones"][0] }}
                    notifTag: '{{ trigger.payload_json["after"]["id"] }}'
                    notifIcon: mdi:doorbell-video
                    group: >-
                      frigate-notification-{{
                      trigger.payload_json["after"]["camera"] }}
                    importance: default
                    image: >-
                      /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                    clickAction: >-
                     BASE URL OF SERVER/api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
    default: []
mode: parallel
max: 10
```

**Comment by NickM-27** [comment]
Okay I tried to get it close. You will need to adjust the camera name, base url of server, and then either setup a script like I have or change that action to notify the device(s) of your choice

**Comment by junior466** [comment]
@NickM-27 Wow sir, you are amazing! Thank you for your time. I will try this shortly!

**Comment by NickM-27** [comment]
Let me know if you have any questions

**Comment by junior466** [comment]
So I tried editing it but don't think I was successful. I tested it by triggering the automation in home assistant but no notification is sent to my phone. I imagine the automation should at least send the notification part to my phone (with no image)? I am also seeing an error in Home Assistant logs:

    Error fetching information from http://192.168.141.21:5000/api/stats: 502, message='Bad Gateway', url=URL('http://192.168.141.21:5000/api/stats')

Here's my edited automation. My zones are `street` for the actual street and `driveway` for the actual driveway. 

    - alias: Frigate - Driveway & Street
      description: ''
      trigger:
        - platform: mqtt
          topic: frigate/events
          id: frigate-event
          payload: driveway_cam
          value_template: '{{ value_json["after"]["camera"] }}'
      condition:
        - condition: or
          conditions:
            - condition: template
              value_template: '{{ trigger.payload_json[''type''] == ''new'' }}'
            - condition: template
              value_template: '{{ trigger.payload_json["before"]["entered_zones"]|length == 0 }}'
        - condition: template
          value_template: '{{ ["street"]|select("in", trigger.payload_json["after"]["entered_zones"])|list|length > 0 and ["driveway"]|select("in", trigger.payload_json["after"]["current_zones"])|list|length > 0 }}'
        - condition: template
          value_template: >-
            {{ ["driveway", "street"]|select("in",
            trigger.payload_json["after"]["entered_zones"])|list|length > 0 }}
        - condition: template
          value_template: '{{ trigger.payload_json["after"]["label"] != "car" }}'
      action:
        - choose:
            - conditions:
                - condition: trigger
                  id: frigate-event
              sequence:
                - service: notify.mobile_app_USER_iphone
                  data:
                    id: '{{ trigger.payload_json["after"]["id"] }}'
                    title: >-
                      {{ trigger.payload_json["after"]["label"] | title }} was
                      detected on {{ trigger.payload_json["after"]["camera"] |
                      replace("_", " ") | title }}
                    message: >-
                      {{ trigger.payload_json["after"]["label"] | title }} detected in
                      the {{ trigger.payload_json["after"]["entered_zones"][0] }}
                    notifTag: '{{ trigger.payload_json["after"]["id"] }}'
                    notifIcon: mdi:doorbell-video
                    group: >-
                      frigate-notification-{{ trigger.payload_json["after"]["camera"]
                      }}
                    importance: max
                    image: >-
                      /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                    clickAction: ''
                - repeat:
                    until:
                      - condition: template
                        value_template: '{{ wait.trigger.payload_json["type"] == "end" }}'
                    sequence:
                      - wait_for_trigger:
                          - platform: mqtt
                            topic: frigate/events
                            payload: '{{ trigger.payload_json["after"]["id"] }}'
                            value_template: '{{ value_json["after"]["id"] }}'
                        continue_on_timeout: false
                        timeout: '00:02:00'
                      - condition: template
                        value_template: '{{ wait.trigger.payload_json[''type''] == ''end'' }}'
                      - service: notify.mobile_app_USER_iphone
                        data:
                          id: '{{ trigger.payload_json["after"]["id"] }}'
                          title: >-
                            {{trigger.payload_json["after"]["label"] | title}} was detected on {{
                            trigger.payload_json["after"]["camera"] | replace("_", "
                            ") | title }}
                          message: >-
                            {{trigger.payload_json["after"]["label"] | title}} detected in the {{
                            trigger.payload_json["after"]["entered_zones"][0] }}
                          notifTag: '{{ trigger.payload_json["after"]["id"] }}'
                          notifIcon: mdi:doorbell-video
                          group: >-
                            frigate-notification-{{
                            trigger.payload_json["after"]["camera"] }}
                          importance: default
                          image: >-
                            /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                          clickAction: >-
                           http://192.168.141.21:5000/api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
          default: []
      mode: parallel
      max: 10

**Comment by NickM-27** [comment]
Oh my bad, you will also want to remove this condition:

```yaml
- condition: template
   value_template: '{{ trigger.payload_json["after"]["label"] != "car" }}'
```

If you suspect something should be triggered you can use the "debug automation" option in HomeAssistant to see the messages that are coming in and which condition it got stopped on.

**Comment by junior466** [comment]
Yea, something is definitely wrong. The logs show no errors and the automation debug looks like this:
![Screen Shot 2022-02-28 at 7 14 10 PM](https://user-images.githubusercontent.com/22748735/156080534-6ab5e6ec-21d7-4316-b539-ce3381022de3.png)

Two questions: Do I need to change `id: frigate-event` to something else? Is the base url I needed to change the URL of Frigate? Just making sure I am not missing anything obvious. I am trying to get to a point where the automation works and at least sends me an alert. Every time I trigger it, nothing pops up on my device.

**Comment by NickM-27** [comment]
Can you click on "Changed Variables" and what that shows? Should be able to quickly see what happened here

**Comment by NickM-27** [comment]
frigate-event is the `TriggerID` which is just a home assistant thing and doesn't affect the MQTT stuff / wouldn't block this from running.

And the base URL is the base URL of frigate

**Comment by junior466** [comment]
I see:

      context:
        id: cbe9813f59e337fa69128626121d84bd
        parent_id: 265f9b146deb3fc7f03dca7849511a61
        user_id: null

**Comment by NickM-27** [comment]
Sorry, click on the yellow icon with arrows going to the left and right (choose) and then check changed_variables

**Comment by junior466** [comment]
![Screen Shot 2022-02-28 at 7 26 06 PM](https://user-images.githubusercontent.com/22748735/156081436-8667cbc7-eb55-4023-92f0-7b3d7b91d6ae.png)

**Comment by NickM-27** [comment]
Just curious, how are you triggering this automation? Are you running manually? This will only work if you post an MQTT messages as just pressing "run actions" will not do anything

**Comment by NickM-27** [comment]
Uhhh well, to test it "end to end" yes. However HomeAssistant does offer the ability to manually trigger MQTT messages so I would start with that just to make sure the notification and other stuff works. One sec I should be able to cook up what you are looking for.

**Comment by NickM-27** [comment]
You will want to use the services section per the screenshot and enter this YAML:

```yaml
service: mqtt.publish
data:
  topic: frigate/events
  payload: >-
    {"before": {"id": "INSERT VALID EVENT ID", "camera": "INSERT CAMERA NAME",
    "frame_time": 1646094818.516091, "snapshot_time": 1646093963.926109,
    "label": "car", "top_score": 0.78125, "false_positive": false, "start_time":
    1646093962.13158, "end_time": null, "score": 0.71484375, "box": [0, 499,
    101, 598], "area": 9999, "region": [0, 364, 320, 684], "stationary": true,
    "motionless_count": 4271, "position_changes": 1, "current_zones": [],
    "entered_zones": [], "has_clip": false, "has_snapshot": false}, "after":
    {"id": "INSERT VALID EVENT ID", "camera": "INSERT CAMERA NAME",
    "frame_time": 1646094878.639409, "snapshot_time": 1646093963.926109,
    "label": "car", "top_score": 0.78125, "false_positive": false, "start_time":
    1646093962.13158, "end_time": null, "score": 0.71484375, "box": [0, 499,
    101, 598], "area": 9999, "region": [0, 364, 320, 684], "stationary": true,
    "motionless_count": 4572, "position_changes": 1, "current_zones": ["driveway"],
    "entered_zones": ["street"], "has_clip": false, "has_snapshot": false}, "type":
    "update"}
```

You will need to change the "id" in both `before` and `after` as well as the `camera` to the name you are expecting. 

![Screen Shot 2022-02-28 at 5 35 31 PM](https://user-images.githubusercontent.com/14866235/156082311-f032c6a3-6904-4c64-9919-672f2211349e.png)

**Comment by junior466** [comment]
Is the `INSERT VALID EVENT ID` the id of the automation?

**Comment by NickM-27** [comment]
That would be the ID of the frigate event. For testing purposes you can use any one you want it is just used for the snapshot

**Comment by junior466** [comment]
I received this:

    Error while executing automation automation.automation_19: extra keys not allowed @ data['id']

**Comment by NickM-27** [comment]
Okay so that is due to the notification part. You edited the service but you are still using my data args. I wrote a custom script so those aren't valid for a standard `notify` call. (BUT HEY, that means it almost worked!)

You will need to change the data part of that notify call to:

```yaml
data:
  title: >-
    {{trigger.payload_json["after"]["label"] | title}} was detected on {{
    trigger.payload_json["after"]["camera"] | replace("_", "
    ") | title }}
  message: >-
    {{trigger.payload_json["after"]["label"] | title}} detected in the {{
    trigger.payload_json["after"]["entered_zones"][0] }}
  data:
    url: >-
      http://192.168.141.21:5000/api/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
    tag: '{{ trigger.payload_json["after"]["id"] }}'
    group: frigate-notification-{{ trigger.payload_json["after"]["camera"] }}
    image: >-
      /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
    attachment:
      url: >-
        /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
    push:
      interruption-level: >-
        {% if importance == "max" -%} time-sensitive {%- else -%} passive {%-
        endif %}
```

**Comment by junior466** [comment]
I know, we are close! Do I remove everything below the notify call `- service: notify.mobile_app_USER_iphone` and add the new code?

      action:
        - choose:
            - conditions:
                - condition: trigger
                  id: frigate-event
              sequence:
                - service: notify.mobile_app_USER_iphone
                  data:
                    id: '{{ trigger.payload_json["after"]["id"] }}'
                    title: >-
                      {{ trigger.payload_json["after"]["label"] | title }} was
                      detected on {{ trigger.payload_json["after"]["camera"] |
                      replace("_", " ") | title }}
                    message: >-
                      {{ trigger.payload_json["after"]["label"] | title }} detected in
                      the {{ trigger.payload_json["after"]["entered_zones"][0] }}
                    notifTag: '{{ trigger.payload_json["after"]["id"] }}'
                    notifIcon: mdi:doorbell-video
                    group: >-
                      frigate-notification-{{ trigger.payload_json["after"]["camera"]
                      }}
                    importance: max
                    image: >-
                      /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                    clickAction: ''
                - repeat:
                    until:
                      - condition: template
                        value_template: '{{ wait.trigger.payload_json["type"] == "end" }}'
                    sequence:
                      - wait_for_trigger:
                          - platform: mqtt
                            topic: frigate/events
                            payload: '{{ trigger.payload_json["after"]["id"] }}'
                            value_template: '{{ value_json["after"]["id"] }}'
                        continue_on_timeout: false
                        timeout: '00:02:00'
                      - condition: template
                        value_template: '{{ wait.trigger.payload_json[''type''] == ''end'' }}'
                      - service: notify.mobile_app_USER_iphone
                        data:
                          id: '{{ trigger.payload_json["after"]["id"] }}'
                          title: >-
                            {{trigger.payload_json["after"]["label"] | title}} was detected on {{
                            trigger.payload_json["after"]["camera"] | replace("_", "
                            ") | title }}
                          message: >-
                            {{trigger.payload_json["after"]["label"] | title}} detected in the {{
                            trigger.payload_json["after"]["entered_zones"][0] }}
                          notifTag: '{{ trigger.payload_json["after"]["id"] }}'
                          notifIcon: mdi:doorbell-video
                          group: >-
                            frigate-notification-{{
                            trigger.payload_json["after"]["camera"] }}
                          importance: default
                          image: >-
                            /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                          clickAction: >-
                           http://192.168.141.21:5000/api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
          default: []
      mode: parallel
      max: 10

**Comment by NickM-27** [comment]
yes, you will want to do on both notify actions

**Comment by junior466** [comment]
Okay, so in the end, I ended up with this and the notification comes through! 👍 

      - id: '164037269411544'
        alias: Frigate - Driveway & Street
        description: ''
        trigger:
          - platform: mqtt
            topic: frigate/events
            id: frigate-event
            payload: driveway_cam
            value_template: '{{ value_json["after"]["camera"] }}'
        condition:
          - condition: or
            conditions:
              - condition: template
                value_template: '{{ trigger.payload_json[''type''] == ''new'' }}'
              - condition: template
                value_template: '{{ trigger.payload_json["before"]["entered_zones"]|length == 0 }}'
          - condition: template
            value_template: '{{ ["street"]|select("in", trigger.payload_json["after"]["entered_zones"])|list|length > 0 and ["driveway"]|select("in", trigger.payload_json["after"]["current_zones"])|list|length > 0 }}'
          - condition: template
            value_template: >-
              {{ ["driveway", "street"]|select("in",
              trigger.payload_json["after"]["entered_zones"])|list|length > 0 }}
          # - condition: template
          #   value_template: '{{ trigger.payload_json["after"]["label"] != "car" }}'
        action:
          - choose:
              - conditions:
                  - condition: trigger
                    id: frigate-event
                sequence:
                  - service: notify.mobile_app_USER_iphone
                    data:
                      title: >-
                        {{trigger.payload_json["after"]["label"] | title}} was detected on {{
                        trigger.payload_json["after"]["camera"] | replace("_", "
                        ") | title }}
                      message: >-
                        {{trigger.payload_json["after"]["label"] | title}} detected in the {{
                        trigger.payload_json["after"]["entered_zones"][0] }}
                      data:
                        url: >-
                          http://192.168.141.21:5000/api/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
                        tag: '{{ trigger.payload_json["after"]["id"] }}'
                        group: frigate-notification-{{ trigger.payload_json["after"]["camera"] }}
                        image: >-
                          /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                        attachment:
                          url: >-
                            /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                  - repeat:
                      until:
                        - condition: template
                          value_template: '{{ wait.trigger.payload_json["type"] == "end" }}'
                      sequence:
                        - wait_for_trigger:
                            - platform: mqtt
                              topic: frigate/events
                              payload: '{{ trigger.payload_json["after"]["id"] }}'
                              value_template: '{{ value_json["after"]["id"] }}'
                          continue_on_timeout: false
                          timeout: '00:02:00'
                        - condition: template
                          value_template: '{{ wait.trigger.payload_json[''type''] == ''end'' }}'
                        - service: notify.mobile_app_USER_iphone
                          data:
                            title: >-
                              {{trigger.payload_json["after"]["label"] | title}} was detected on {{
                              trigger.payload_json["after"]["camera"] | replace("_", "
                              ") | title }}
                            message: >-
                              {{trigger.payload_json["after"]["label"] | title}} detected in the {{
                              trigger.payload_json["after"]["entered_zones"][0] }}
                            data:
                              url: >-
                                http://192.168.141.21:5000/api/{{trigger.payload_json["after"]["id"]}}/{{trigger.payload_json["after"]["camera"]}}/clip.mp4
                              tag: '{{ trigger.payload_json["after"]["id"] }}'
                              group: frigate-notification-{{ trigger.payload_json["after"]["camera"] }}
                              image: >-
                                /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
                              attachment:
                                url: >-
                                  /api/frigate/notifications/{{trigger.payload_json["after"]["id"]}}/snapshot.jpg
            default: []
        mode: parallel
        max: 10

**Comment by NickM-27** [comment]
Awesome! Glad to see it. Hopefully that will be a clean-ish starting point so you can modify from there to your specific needs

**Comment by junior466** [comment]
Yes, definitely! Can't wait to actually try it out tomorrow! One last thing since you are so experienced! Does this configuration look right to only capture events that enter the driveway? My street is quite busy so I only want to capture cars that enter the driveway just like the notification. 

##########################################################
## Driveway
##########################################################

cameras:
  driveway_cam:
    rtmp:
      enabled: False

    ffmpeg:
      inputs:
        - path: rtsp://{FRIGATE_RTSP_USER}:{FRIGATE_RTSP_PASSWORD}@192.168.141.14:554/cam/realmonitor?channel=1&subtype=0
          roles:
            - detect
            - record

    detect:
      width: 3840
      height: 2160
      fps: 5
      enabled: True

    zones:
      driveway:
        coordinates: 2659,2160,2110,1912,1885,1401,2381,1215,3813,1502,3840,2160
        objects:
          - car
          - person
          - dog
      street:
        coordinates: 3840,1088,2617,902,2280,1146,3840,1533
        objects:
          - car


    objects:
      track:
        - person
        - dog
        - car
      filters:
        person:
          # Optional: minimum width*height of the bounding box for the detected object (default: 0)
          min_area: 5000
          # Optional: maximum width*height of the bounding box for the detected object (default: 24000000)
          max_area: 100000
          mask:
            - 0,0,0,2160,2098,2160,1850,1401,2714,914,3840,1080,3840,0
        car:
          # Optional: minimum width*height of the bounding box for the detected object (default: 0)
          min_area: 5000
          # Optional: maximum width*height of the bounding box for the detected object (default: 24000000)
          max_area: 200000
          mask:
            - 0,0,0,2160,2098,2160,1850,1401,2714,914,3840,1080,3840,0
        dog:
          mask:
            - 0,0,0,2160,2098,2160,1850,1401,2714,914,3840,1080,3840,0

    motion:
      # Increasing this value will make motion detection less sensitive
      # threshold: 30
      # default 25

      # Minimum size in pixels in the resized motion image that counts as motion.
      # Increasing this value will prevent smaller areas of motion from being detected. Decreasing will
      # make motion detection more sensitive to smaller moving objects.
      # As a rule of thumb:
      #  - 15 - high sensitivity
      #  - 30 - medium sensitivity
      #  - 50 - low sensitivity
      contour_area: 35
      # default: 30
      mask:
       - 0,0,0,2160,2098,2160,1850,1401,2714,914,3840,1080,3840,0

    record:
      enabled: True
      retain:
        days: 7
        mode: motion
      events:
        max_seconds: 240
        pre_capture: 5
        post_capture: 5
        retain:
          default: 7
          mode: active_objects
        objects:
          - person
          - car
          - dog
        required_zones:
          - driveway

    snapshots:
      enabled: True
      timestamp: True
      bounding_box: True
      crop: False
      required_zones:
        - driveway
      retain:
        default: 7

#########################################################
## Global
#########################################################

mqtt:
  host: 192.168.141.21
  user: USER
  password: PASSWORD

detectors:
  coral:
    type: edgetpu
    device: usb

database:
  path: /db/frigate.db

snapshots:
  enabled: True
  timestamp: True
  bounding_box: True
  crop: False
  required_zones: []
  retain:
    default: 7

model:
  labelmap:
    15: animal
    16: animal
    17: animal
    18: animal
    19: animal
    20: animal
    21: animal
    22: animal
    23: animal
    24: animal

birdseye:
  enabled: False
  # width: 1920
  # height: 1080
  # quality: 8
  # mode: motion

**Comment by NickM-27** [comment]
As far as the zones go that looks right. It will only create events for cars (person + animal as well) that are in the driveway. And the street notification will be used to only show notifications when that happens. 

One thing to note though, you have the labelmap putting a bunch of labels to `animal` but you are still using `dog` in the zones and objects settings of your camera which will actually do nothing since that also needs to say `animal` (dog doesn't exist anymore since it is now called animal so dog won't even work)

**Comment by junior466** [comment]
> And the street zone will be used to only show notifications when that happens.

You mean when a car is detected in the street zone and then ENTERS the driveway zone an event will be created and a notification sent? 

> One thing to note though, you have the labelmap putting a bunch of labels to animal but you are still using dog in the zones and objects settings of your camera which will actually do nothing since that also needs to say animal (dog doesn't exist anymore since it is now called animal so even dog won't work)

That was a mistake on my part. Thanks for pointing it out.

**Comment by NickM-27** [comment]
So the behaviors will be different for event and notification. An event will be created for the car while it is in the driveway. 

A notification will be sent once the car enters the driveway after having been in the street.

**Comment by blakeblackshear** [comment]
I don't think you should get events for objects that never change position, so if there is a parked car, it shouldn't create an event if it never moves. It may get confusing while tracking if a car drives by, but generally this should be the case.

**Comment by junior466** [comment]
> So the behaviors will be different for event and notification. An event will be created for the car while it is in the driveway.
> 
> A notification will be sent once the car enters the driveway after having been in the street.

Awesome! Thank you very much for the step by step assistance! I really appreciate it. Can’t wait to try it tomorrow.

**Comment by junior466** [comment]
So after trying it out nearly all day (I wasn't home but wife was in and out of the driveway) no notification was ever received. Not sure what went wrong but based on our test, that event triggered a notification. Not looking to take more of your time on this issue and I may just end up only detecting `person` since its proving more and more my personal setup.

**Comment by NickM-27** [comment]
I would recommend checking the debug automation section again and see what the values are. Probably some condition that isn't quite right

**Comment by junior466** [comment]
One question I do have. What part of this automation is actually responsible for triggering the notification only after a car has entered the street AND driveway zone? If a car is parked on the driveway and one drives by the `street` zone, wouldn't it trigger the automation (which is something I don't want)? Again, just trying to make sure that what I am trying to do is actually possible.

**Comment by NickM-27** [comment]
It's the condition that checks for both street and driveway zones. Frigate has event IDs for each object so it's not like a car that drives by will trigger it if there's a car in the driveway. It has to be the same car.

**Comment by junior466** [comment]
Ohh I see. Well, that's exactly what I need them. Should I look at the debug information only when I think that the event happened (i.e. car on the street zone that entered the driveway)? And how come that service call (mqtt.publish) you gave me to try was able to at least push a notification?
